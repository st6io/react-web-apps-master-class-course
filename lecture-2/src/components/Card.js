import React from 'react';

import styles from './common.module.scss';

const Card = ({ children, className }) => (
  <div className={`${styles.card} ${className || ''}`}>{children}</div>
);

export default Card;
