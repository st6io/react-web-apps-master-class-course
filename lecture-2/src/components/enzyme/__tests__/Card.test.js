import React from 'react';
import { shallow } from 'enzyme';

import Card from '../../Card';

describe('Card Component', () => {
  it('should match snapshot', () => {
    const wrapper = shallow(
      <Card>
        <span>Hello, World!</span>
      </Card>,
    );

    expect(wrapper).toMatchSnapshot();
  });

  it('should match snapshot with className', () => {
    const wrapper = shallow(
      <Card className="new-class-name">
        <span>Hello, World!</span>
      </Card>,
    );

    expect(wrapper).toMatchSnapshot();
  });
});
