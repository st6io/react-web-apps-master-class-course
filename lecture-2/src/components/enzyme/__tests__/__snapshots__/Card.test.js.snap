// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card Component should match snapshot 1`] = `
<div
  className="card "
>
  <span>
    Hello, World!
  </span>
</div>
`;

exports[`Card Component should match snapshot with className 1`] = `
<div
  className="card new-class-name"
>
  <span>
    Hello, World!
  </span>
</div>
`;
